### ISSUE TITLE
[CDCSDK] [PG Parity] Snapshot of table from PG connector failed with 'ERROR: Snapshot too old.'

### ISSUE THREAD

**Issue by shamanthchandra-yb** [yugabyte/yugabyte-db#22129]
Jira Link: [DB-11054](https://yugabyte.atlassian.net/browse/DB-11054)
### Description

Please find stress run in JIRA.

**Steps:**

- Restoring data to Yugabyte Database of about 5000000 rows.
- Deploy PG connector with initial mode.
- Verify data in sink target
- Some nemesis has been running in parallel

![Profile (11)](https://github.com/yugabyte/yugabyte-db/assets/109518123/9b8ea6c8-73ee-48a4-8cf7-fee71e31e4b6)

```
9899 2024-04-24 08:22:50,291 ERROR  Postgres|db_cdc|snapshot  Error during snapshot   [io.debezium.relational.RelationalSnapshotChangeEventSource]
 9900 java.util.concurrent.ExecutionException: org.apache.kafka.connect.errors.ConnectException: Snapshotting of table public.test_cdc_e855d7 failed
 9901         at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
 9902         at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
 9903         at io.debezium.relational.RelationalSnapshotChangeEventSource.createDataEvents(RelationalSnapshotChangeEventSource.java:468)
 9904         at io.debezium.relational.RelationalSnapshotChangeEventSource.doExecute(RelationalSnapshotChangeEventSource.java:165)
 9905         at io.debezium.pipeline.source.AbstractSnapshotChangeEventSource.execute(AbstractSnapshotChangeEventSource.java:92)
 9906         at io.debezium.pipeline.ChangeEventSourceCoordinator.doSnapshot(ChangeEventSourceCoordinator.java:250)
 9907         at io.debezium.pipeline.ChangeEventSourceCoordinator.doSnapshot(ChangeEventSourceCoordinator.java:234)
 9908         at io.debezium.pipeline.ChangeEventSourceCoordinator.executeChangeEventSources(ChangeEventSourceCoordinator.java:186)
 9909         at io.debezium.pipeline.ChangeEventSourceCoordinator.lambda$start$0(ChangeEventSourceCoordinator.java:137)
 9910         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
 9911         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 9912         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 9913         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 9914         at java.base/java.lang.Thread.run(Thread.java:829)
 9915 Caused by: org.apache.kafka.connect.errors.ConnectException: Snapshotting of table public.test_cdc_e855d7 failed
 9916         at io.debezium.relational.RelationalSnapshotChangeEventSource.doCreateDataEventsForTable(RelationalSnapshotChangeEventSource.java:598)
 9917         at io.debezium.relational.RelationalSnapshotChangeEventSource.lambda$createDataEventsForTableCallable$6(RelationalSnapshotChangeEventSource.java:520)
 9918         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 9919         ... 5 more
 9920 Caused by: com.yugabyte.util.PSQLException: ERROR: Snapshot too old. Read point: { physical: 1713945775311466 }, earliest read time allowed: { physical: 1713946070018055 }, delta (usec): 294706589: kSnapshotTooOld
 9921         at com.yugabyte.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2675)
 9922         at com.yugabyte.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2365)
 9923         at com.yugabyte.core.v3.QueryExecutorImpl.fetch(QueryExecutorImpl.java:2561)
 9924         at com.yugabyte.jdbc.PgResultSet.next(PgResultSet.java:2087)
 9925         at io.debezium.jdbc.CancellableResultSet.next(CancellableResultSet.java:52)
 9926         at io.debezium.relational.RelationalSnapshotChangeEventSource.doCreateDataEventsForTable(RelationalSnapshotChangeEventSource.java:577)
 9927         ... 7 more
```

### Source connector version

`fourpointfour/ybdb-debezium:0.5`

### Connector configuration

```
add connector connector_name='ybconnector_x_24c5ea_test_cdc_e855d7' stream_id='cda6662b4ccbccb6104fff1e945fd82d' db_name='x_24c5ea' connector_host='172.151.17.241' table_list=['test_cdc_e855d7'] {'name': 'ybconnector_x_24c5ea_test_cdc_e855d7', 'config': {'database.master.addresses': '172.151.18.37:7100,172.151.25.92:7100,172.151.28.89:7100', 'database.hostname': '172.151.18.37:5433,172.151.25.92:5433,172.151.28.89:5433', 'database.port': 5433, 'database.masterhost': '172.151.18.37', 'database.masterport': '7100', 'database.user': 'yugabyte', 'database.password': 'yugabyte', 'database.dbname': 'x_24c5ea', 'snapshot.mode': 'initial', 'admin.operation.timeout.ms': 600000, 'socket.read.timeout.ms': 300000, 'max.connector.retries': '10', 'operation.timeout.ms': 600000, 'topic.creation.default.compression.type': 'lz4', 'topic.creation.default.cleanup.policy': 'delete', 'topic.creation.default.partitions': 2, 'topic.creation.default.replication.factor': '1', 'tasks.max': '5', 'connector.class': 'io.debezium.connector.postgresql.PostgresConnector', 'topic.prefix': 'db_cdc', 'plugin.name': 'pgoutput', 'slot.name': 'cda6662b4ccbccb6104fff1e945fd82d_from_con', 'publication.autocreate.mode': 'filtered', 'publication.name': 'pn_ybconnector_x_24c5ea_test_cdc_e855d7', 'table.include.list': 'public.test_cdc_e855d7', 'max.poll.interval.ms': '5000'}}
```

### YugabyteDB version

2.23.0.0-b207

### Issue Type

kind/bug

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-11054]: https://yugabyte.atlassian.net/browse/DB-11054?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by strahe** [comment]
I had the same issue. How can I find out if this bug has been resolved?

**Comment by shamanthchandra-yb** [comment]
@strahe this was seen in our chaos testing experiments with logical replication feature of CDC.

Could you add few details regarding your case:
1. Is it you are trying to get 'initial snapshot' using CDC ? If yes, what is the size of the table ?
2. Are there any node restarts or any kind of chaos/nemesis happening on your server side, when this was running ?

cc: @suranjan

**Comment by strahe** [comment]
1. Yes, this table is used to record real-time tasks. A record is added when a task starts and deleted when the task is completed. The table is very small or even empty.
2. No, I only ran a single node, and it works fine.

**Comment by strahe** [comment]
I changed the snapshot mode to never, and the issue no longer occurs. but I had another issue.

![image](https://github.com/user-attachments/assets/bd508d27-088e-4a5a-ac6b-349fe083d814)

I'm not sure where this table comes from, as it doesn't actually exist in the database.

**Comment by shamanthchandra-yb** [comment]
@strahe was there any DDL op performed ? which could be causing table rewrite ?

If yes, can you try creating new replication slot and check.

**Comment by strahe** [comment]
There are no operations that would cause a table rewrite, even if I create a new replication slot, the same issue occurs.

![image](https://github.com/user-attachments/assets/a697a9ac-7a7a-4b2e-a502-c7172c5e4d3c)

**Comment by suranjan** [comment]
@strahe Could you please let us know the version of YugabyteDB? This has been fixed in master.

**Comment by strahe** [comment]
Glad to know this news, I am using version 2024.1.2.0-b77.
